File: Python/Memory.py
from Hardware import Hardware

class Memory(Hardware):
    
    # constructor
    def __init__(self):
        self.frequency = "blank"
        self.memorySize = 0
        self.supportsCuda = "blank"
        
    # methods for frequency attribute
    def setFrequency(self, frequency):
        self.frequency = frequency
        
    def getFrequency(self):
        return self.frequency 

    # methods for memorys size attribute
    def setMemorySize(self, memorySize):
        self.memorySize = memorySize
        
    def getMemorySize(self):
        return self.memorySize 

    # methods for supports cuda attribute
    def setSupportsCuda(self, supportsCuda):
        self.supportsCuda = supportsCuda
        
    def getSupportsCuda(self):
        return self.supportsCuda 

File: PHP/Memory.php
<?php 

    // class for Memory
    class Memory extends Hardware
    {
        // public attributes
        public $frequency;
        public $memorysize;
        public $cuda;

        // constructor
        public function __construct($frequency = "", $memorysize = "", $cuda = "")
        {
            $this->frequency = $frequency;
            $this->memorysize = $memorysize;
            $this->cuda = $cuda;            
        }

        // methods for frequency attribute
        public function setFrequency($frequency)
        {
            $this->frequency = $frequency;
        }   
        public function getFrequency()
        {
            return $this->frequency;
        }

        // methods for memory size attribute
        public function setMemorySize($memorysize)
        {
            $this->memorysize = $memorysize;
        }   
        public function getMemorySize()
        {
            return $this->memorysize;
        }
        
        // methods for supports cuda attribute
        public function setCuda($cuda)
        {
            $this->cuda = $cuda;
        }   
        public function getCuda()
        {
            return $this->cuda;
        }

        // destructor
        public function __destruct()
        {
            
        }
    }
?>